locallib.php

```diff
@@ -533,24 +576,126 @@ function save_attempt($assignment, $filename, $evaluation, $recordinglength) {
+        $feedback->lexicalprofile_reason = $fromform->lexicalprofilereason;
+
+        $feedback->old_nativeity = $oldattempt->nativeity;
+        $feedback->nativeity = $fromform->nativeity;
+        $feedback->nativeity_reason = $fromform->nativeityreason;
+
+        $feedback->old_holistic = $oldattempt->holistic;
+        $feedback->holistic = $fromform->holistic;
+        $feedback->holistic_reason = $fromform->holisticreason;
+
+    } else if ($attempttype == 'readaloud') {
+        $feedback->old_gop_score = $oldattempt->gop_score;
+        $feedback->gop_score = $fromform->gop;
+        $feedback->gop_score_reason = $fromform->gopreason;
+    }
+
+    $timenow = time();
+
+    $feedback->timecreated = $timenow;
+
+    $DB->insert_record('digitala_report_feedback', $feedback);
+}
+
 /**
  * Load current users attempt from the database.
  *
  * @param int $instanceid - instance id of this digitala activity
+ * @param int $userid - user id of this user or student
  * @return mixed $attempt - object containing attempt information
  */
-function get_attempt($instanceid) {
-    global $DB, $USER;
+function get_attempt($instanceid, $userid) {
+    global $DB;
 
-    if (!$DB->record_exists('digitala_attempts', array('digitala' => $instanceid, 'userid' => $USER->id))) {
+    if (!$DB->record_exists('digitala_attempts', array('digitala' => $instanceid, 'userid' => $userid))) {
         return null;
     }
 
-    $attempt = $DB->get_record('digitala_attempts', array('digitala' => $instanceid, 'userid' => $USER->id));
+    $attempt = $DB->get_record('digitala_attempts', array('digitala' => $instanceid, 'userid' => $userid));
 
     return $attempt;
 }
 
+/**
+ * Load all attempts from the database.
+ *
+ * @param int $instanceid - instance id of this digitala activity
+ * @return $attempts - object containing all attempt information
+ */
+function get_all_attempts($instanceid) {
+    global $DB;
+
+    $attempts = $DB->get_records('digitala_attempts', array('digitala'  => $instanceid));
+    return $attempts;
+}
+
+/**
+ * Load users name based on their id.
+ *
+ * @param int $id - id of the user
+ * @return $user - user object
+ */
+function get_user($id) {
+    global $DB;
+
+    $user = $DB->get_record('user', array('id' => $id));
+    return $user;
+}
+
+/**
+ * Load all attempts from the database.
+ *
+ * @param mixed $attempt - object containing attempt information
+ * @param int $instanceid - instance id of this digitala activity
+ * @param int $id - activity id
+ * @param int $d - course id
+ * @return $cells - cells containing table data
+ */
+function create_result_row($attempt, $instanceid, $id, $d) {
+    $user = get_user($attempt->userid);
+
+    $username = $user->firstname . ' ' . $user->lastname;
+    if ($attempt->holistic) {
+        $score = $attempt->holistic;
+    } else {
+        $score = $attempt->gop_score;
+    }
+    $time = ($attempt->recordinglength . ' s');
```

Comment:
```suggestion
    $time = get_($attempt->recordinglength);
```
Does converting this to format `02:27` (2 minutes and 27 seconds) make more sense than showing `147 s`?

Reply:
It does look a lot better, will commit this suggestion.